staging/src/k8s.io/client-go/tools/cache/delta_fifo.go

```diff
@@ -57,14 +57,53 @@ import (
 //
 // Also see the comment on DeltaFIFO.
 func NewDeltaFIFO(keyFunc KeyFunc, knownObjects KeyListerGetter) *DeltaFIFO {
+	return NewDeltaFIFOWithOptions(DeltaFIFOOptions{
+		KeyFunction:  keyFunc,
+		KnownObjects: knownObjects,
+	})
+}
+
+// DeltaFIFOOptions is the configuration parameters for DeltaFIFO. All are
+// optional.
+type DeltaFIFOOptions struct {
+
+	// KeyFunction is used to figure out what key an object should have. (It's
+	// exposed in the returned DeltaFIFO's KeyOf() method, with bonus features.)
+	// Optional, the default is MetaNamespaceKeyFunc which is essentially
+	// <namespace>/<name>
+	KeyFunction KeyFunc
+
+	// KnownObjects is expected to return a list of keys that the consumer of
+	// this queue "knows about". It is used to decide which items are missing
+	// when Replace() is called; 'Deleted' deltas are produced for these items.
+	// It may be nil if you don't need to detect all deletions.
+	KnownObjects KeyListerGetter
+
+	// ClientUnderstandsReplaced is whether or not the queue consumer understands
+	// the Replaced DeltaType. Before Replaced was added, Replace() was treated
+	// the same as Sync(). For backwards-compatability purposes, this is false
+	// by default.
+	ClientUnderstandsReplaced bool
```

Comment:
This approach lgtm.

nit: I would be more explicit about the name (e.g. `ClientUnderstandsReplacedDeltaType`)